### ISSUE TITLE
Slow5tools merge failed 

### ISSUE THREAD

**Issue by NStrowbridge** [hasindu2008/slow5tools#135]
Hi @hasindu2008 , 

I have been recently trying to create an eventalign file for some Nanopore Direct RNA data and have come up with an error as follows: Failed to allocate memory: Cannot allocate memory At src/slow5_idx.c:280

I went on the f5c github and noticed a very similar issue from June of this year. I tried all the advice on there, including downloading the test dataset to test slow5tools index and quickcheck. The quickcheck worked for my dataset, and index worked for the test dataset. 

As such, I then tried to use slow5tools merge on my resulting blow5 files from: blue-crab p2s pod5 -d blow5

I have come up with the following error message: 

[slow5_hdr_init::ERROR] Malformed blow5 header. Invalid magic number. At src/slow5.c:806
[slow5_init::ERROR] Parsing slow5 header of file 'blow5/._FAZ32689_11a0f377_3f7114be_0.blow5' failed. At src/slow5.c:155
[slow5_open_with::ERROR] Exiting on error. At src/slow5.c:371

I was wondering if you could provide some advice on how to proceed. This is not a publicly assessed dataset, but my own. 

Oh, I also should mention that this is RNA004 data, so the raw data are pod5 files. 

Kind regards, 

Nic

**Comment by NStrowbridge** [comment]
Oh, I should mention that this is RNA004 data, so the raw dats are pod5 files.

**Comment by hasindu2008** [comment]
Hey
Could you do the following:
Under the pod5 directory, there should be a file like `FAZ32689_11a0f377_3f7114be_0.pod5`.
Could you do `blue-crab p2s FAZ32689_11a0f377_3f7114be_0.pod5 -o FAZ32689_11a0f377_3f7114be_0.blow5` and then `slow5tools view FAZ32689_11a0f377_3f7114be_0.blow5 -o FAZ32689_11a0f377_3f7114be_0.slow5`?

**Comment by NStrowbridge** [comment]
@hasindu2008 I did what you suggested and it seemed to work. I also ran index on the single slow5 file and that seemed to work also. 

So would you suggest running conversion to slow5 of all blow5 files and then merging? 

The slow5 file is considerably larger than both blow5 and pod5 file: 14GB vs ~3.1GB

**Comment by hasindu2008** [comment]
Hey, slow5 is ASCII, only for human readability and not optimal for size and performance. This test was to see if that particular file had an issue, but seems not. I guess something may have happened during the conversion earlier, such as the job getting killed/running out of disk space or something. 

1. Can you see if all the files under `blow5/` have this thing like `blow5/._FAZ32689_11a0f377_3f7114be_0.blow5` with this '.' rather than being `blow5/FAZ32689_11a0f377_3f7114be_0.blow5`?
2. what does your pod5 directory listing look like? Do they also have this weird `._` thing for each pod5? Do they have directories inside them and have files with the same name under each directory?
3. Could you rerun the blue-crab on the whole dataset and try merging again - to see if this error happens again in the same file?

**Comment by Psy-Fer** [comment]
Hey,

With blue-crab, you can run it a few ways. If your goal is to get 1 big blow5 file, there is another way to skip the merging step with slow5tools, and potentially highlight any other issues that may have happened.

So you should be able to do something like:

```
 blue-crab p2s pod5 ./data.blow5
```

where `pod5` is a dir and `data.blow` is a single file output. You can use `--threads` to handle how many threads to use in the slow5 compression if you want to speed the process up a bit.

Let me know if you have any troubles.

Cheers,
James

**Comment by NStrowbridge** [comment]
Hi all, thanks for the responses! I stepped away from this for a bit but am now back trying things. My files didn't seem to have "._" before them, I'm not sure why the error said that. However, I did do the recommended 

`blue-crab p2s pod5 data.blow5`

And this worked on my first sample, I was able to index and eventalign my files! As you said @hasindu2008 there must've been an issue with my original conversion such as it getting killed/running out of disk space or something. I'm working through my other samples at the moment, I will let you know if I encounter any issues.
